### PR TITLE
contrib/gocql: add NewCluster function to trace clients

### PR DESCRIPTION
**Pull request by rarguelloF**
### What does this PR do?

Adds `db.cassandra.contact.points` to `contrib/gocql`.

- Adds `gocqltrace.NewCluster` method to wrap the Cassandra client since it's created instead of individual queries.
- Deprecate current `WrapQuery` and `WrapBatch` in favor of the new `NewCluster` method. 
- Updates the example to encourage new users to use this method instead.
- Adds `db.cassandra.contact.points` tag for clients traced with `NewCluster`.

### Motivation

We want to be able to tag `db.cassandra.contact.points` (and probably other stuff from the config in the future). With the current way of tracing this integration (`WrapQuery` and `WrapBatch`), we don't have access to the information that was specified during the client initialization unless we use reflection, that we should avoid.

### Reviewer's Checklist

- [x] Changed code has unit tests for its functionality.
- [x] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

File: contrib/gocql/gocql/gocql.go [hunk 1]
```diff
@@ -284,3 +303,38 @@ func (tb *Batch) finishSpan(span ddtrace.Span, err error) {
...
+	s, ok := sv.Interface().(*gocql.Session)
+	if !ok {
+		return nil, false
+	}
+	cfg, ok := getClusterConfig(s)
+	if !ok {
+		return nil, false
+	}
+	return cfg.Hosts, true
+}
```
Comment (ahmed-mez) [inline contrib/gocql/gocql/gocql.go:328]: Isn't this the same reason why we didn't merge https://github.com/DataDog/dd-trace-go/pull/1765 ?

Comment (rarguelloF) [reply contrib/gocql/gocql/gocql.go:328]: @ahmed-mez yes! don't bother about this PR for now, I'm planning to setup a discussion offline about considering the usage of reflection in contribs (we can maybe use this PR as an example use-case). I'll move back this PR to draft for now.

File: contrib/gocql/gocql/gocql.go [hunk 2]
```diff
@@ -28,23 +28,49 @@ func init() {
 	telemetry.LoadIntegration(componentName)
 }
 
+type ClusterConfig struct {
```
Comment (ajgajg1134) [inline contrib/gocql/gocql/gocql.go:31]: Please add godoc comments for all these new public and exported types and functions

File: contrib/gocql/gocql/gocql.go [hunk 3]
```diff
@@ -237,6 +310,14 @@ func (tb *Batch) WithContext(ctx context.Context) *Batch {
 	return tb
 }
 
+// WithWrapOptions allows to specify options specific to the batch.
```
Comment (ajgajg1134) [inline contrib/gocql/gocql/gocql.go:313]:
```suggestion
// WithWrapOptions specifies options specific to the batch.
```

File: contrib/gocql/gocql/gocql.go [hunk 4]
```diff
@@ -93,6 +134,14 @@ func (tq *Query) WithContext(ctx context.Context) *Query {
 	return tq
 }
 
+// WithWrapOptions allows to specify options specific to the query.
```
Comment (ajgajg1134) [inline contrib/gocql/gocql/gocql.go:137]:
```suggestion
// WithWrapOptions specifies options specific to the query.
```

File: contrib/gocql/gocql/gocql_test.go [hunk 5]
```diff
@@ -362,6 +375,90 @@ func TestBatch(t *testing.T) {
...
+	defer mt.Stop()
+
+	cluster := NewCluster([]string{cassandraHost, "127.0.0.1:9043"})
+	updateTestClusterConfig(cluster.ClusterConfig)
+
+	session, err := cluster.CreateSession()
+	require.NoError(t, err)
+	q := session.Query("CREATE KEYSPACE trace WITH REPLICATION = { 'class' : 'NetworkTopologyStrategy', 'datacenter1' : 1 };")
+	err = q.Iter().Close()
+	require.Error(t, err)
```
Comment (ajgajg1134) [inline contrib/gocql/gocql/gocql_test.go:393]: This requires an error?

File: contrib/gocql/gocql/gocql_test.go [hunk 6]
```diff
@@ -362,6 +375,90 @@ func TestBatch(t *testing.T) {
...
+	defer mt.Stop()
+
+	cluster := newTracedCassandraCluster(WithServiceName("test-service"), WithResourceName("cluster-resource"))
+
+	session, err := cluster.CreateSession()
+	require.NoError(t, err)
+	q := session.Query("CREATE KEYSPACE trace WITH REPLICATION = { 'class' : 'NetworkTopologyStrategy', 'datacenter1' : 1 };")
+	q = q.WithWrapOptions(WithResourceName("test-resource"))
+	err = q.Iter().Close()
+	require.Error(t, err)
```
Comment (ajgajg1134) [inline contrib/gocql/gocql/gocql_test.go:431]: same here?

**Review by ajgajg1134 — CHANGES_REQUESTED** [review]

File: contrib/gocql/gocql/gocql_test.go (hunk 5 above)
Comment (rarguelloF) [reply contrib/gocql/gocql/gocql_test.go:393]: added `IF NOT EXISTS` to the query so an error is not expected `require.NoError`

File: contrib/gocql/gocql/gocql_test.go (hunk 6 above)
Comment (rarguelloF) [reply contrib/gocql/gocql/gocql_test.go:431]: did same change here

File: contrib/gocql/gocql/gocql.go [hunk 7]
```diff
@@ -93,6 +139,14 @@ func (tq *Query) WithContext(ctx context.Context) *Query {
 	return tq
 }
 
+// WithWrapOptions specifies options specific to the query.
```
Comment (dianashevchenko) [inline contrib/gocql/gocql/gocql.go:142]:
```suggestion
// WithWrapOptions applies the given set of options to the query.
```

File: contrib/gocql/gocql/gocql.go [hunk 8]
```diff
@@ -237,6 +315,14 @@ func (tb *Batch) WithContext(ctx context.Context) *Batch {
 	return tb
 }
 
+// WithWrapOptions specifies options specific to the batch.
```
Comment (dianashevchenko) [inline contrib/gocql/gocql/gocql.go:318]: similar here, `applies the given set of options to the `

**Pull request closed without merging**